containers/dgxie/api.py

```diff
@@ -2,19 +2,32 @@
 from flask import Flask, abort, request
 import json
 import datetime
+import re
+import os
 
 app = Flask(__name__)
 
 @app.route('/v1/boot/<mac>')
 def pxe(mac):
+    '''See https://github.com/danderson/netboot/blob/master/pixiecore/README.api.md'''
     # load machine profiles for each call so we can re-load changes from disk
     jf = open('/etc/machines/machines.json', 'r')
     machines = json.load(jf)
     jf.close()
 
+    if "HTTP_PORT" in os.environ.keys():
+        http_port = os.environ['HTTP_PORT']
+    else:
+        http_port = "80"
+
     # return profile in json for matching machine
     for machine in machines:
-        if 'mac' in machines[machine] and machines[machine]['mac'] == mac:
+        if 'mac' in machines[machine] and re.match(machines[machine]['mac'], mac):
+            machines[machine]['mac'] = mac
+            machines[machine]['kernel'] = machines[machine]['kernel'].replace("$HTTP_PORT", http_port)
+            machines[machine]['cmdline'] = machines[machine]['cmdline'].replace("$HTTP_PORT", http_port)
```

Comment:
According to the API spec (https://github.com/danderson/netboot/blob/master/pixiecore/README.api.md) only the kernel is a required parameter, but this requires cmdline and initrd to be specified or the API server fails